packages/react/src/date-picker/Calendar.tsx

```diff
@@ -76,8 +109,80 @@ export function CalendarRange({
 const currentYear = getYear(new Date());
 
 const calendarComponents: CustomComponents = {
-	IconLeft: () => <ChevronLeftIcon color="inherit" weight="bold" />,
-	IconRight: () => <ChevronRightIcon color="inherit" weight="bold" />,
+	/*
+		Custom `Caption` component to improve acessibility and focus order of react-day-picker
+		Default:
+		* Caption.tsx: https://github.com/gpbl/react-day-picker/blob/9ad13dc72fff814dcf720a62f6e3b5ea38e8af6d/src/components/Caption.tsx
+		* CaptionNavigation.tsx: https://github.com/gpbl/react-day-picker/blob/9ad13dc72fff814dcf720a62f6e3b5ea38e8af6d/src/components/CaptionNavigation.tsx
+		* Navigation.tsx: https://github.com/gpbl/react-day-picker/blob/9ad13dc72fff814dcf720a62f6e3b5ea38e8af6d/src/components/Navigation.tsx
+	*/
+	Caption: function Caption(props: CaptionProps) {
+		const {
+			classNames,
+			styles,
+			components,
+			labels: { labelPrevious, labelNext },
+			locale,
+		} = useDayPicker();
+		const { previousMonth, nextMonth, goToMonth } = useNavigation();
+
+		const handlePreviousClick: MouseEventHandler = () => {
+			if (!previousMonth) return;
+			goToMonth(previousMonth);
+		};
+
+		const handleNextClick: MouseEventHandler = () => {
+			if (!nextMonth) return;
+			goToMonth(nextMonth);
+		};
+
+		const CaptionLabelComponent = components?.CaptionLabel;
+
+		const previousLabel = labelPrevious(previousMonth, { locale });
+		const previousClassName = [
+			classNames.button_reset,
+			classNames.button,
+			classNames.nav_button,
+			classNames.nav_button_previous,
+		].join(' ');
+
+		const nextLabel = labelNext(nextMonth, { locale });
+		const nextClassName = [
+			classNames.button_reset,
+			classNames.button,
+			classNames.nav_button,
+			classNames.nav_button_next,
+		].join(' ');
+
+		return (
+			<div className={classNames.caption} style={styles.caption}>
+				<button
+					aria-label={previousLabel}
+					className={previousClassName}
+					disabled={!previousMonth}
+					onClick={handlePreviousClick}
+					type="button"
+				>
+					<ChevronLeftIcon color="inherit" weight="bold" />
+				</button>
+				{/* @ts-expect-error: JSX element type 'CaptionLabelComponent' does not have any construct or call signatures. */}
```

Comment:
@ChrisLaneAU Is there anything I can do about this one? It feels easier to solve

Reply:
Looks like the issue is that `CaptionLabelComponent` could be `undefined`. You could conditionally render it and safely remove the `ts-expect-error`.

Reply:
🙏